macros/sql_gen/gen_joined_metrics_cte.sql

```diff
@@ -236,12 +254,31 @@
     joined_metrics as (
 
         select 
-            first_join_metrics.*
```

Comment:
What columns did you have to ditch that meant this can't be a `*` anymore? I'm pricking my ears up at all this select statement logic being replicated. (I doubt it needs to be pulled it out into a macro, that would probably be overkill, but I'm wondering _why_)

Reply:
This was because of the DIV/0 issues. Ideally I could have done `select * except fields we overwrote 0 to null` but except doesn't' exist in all DWHs. With the `select *` it wouldn't allow me to re-overwrite the nulls with 0s. 

Obviously this is kind of gross behaviour to have to re-select everything but the alternative method I thought of had the same issue (removing the coalesce,0 in original metric calc and allowing null values then overwriting in this macro). We'd still have to select with those loops and then do the same thing at the bottom because we don't have select except.